modules/carousel/jetpack-carousel.php

```diff
@@ -545,13 +548,15 @@ function get_attachment_comments() {
 	function post_attachment_comment() {
 		if ( ! headers_sent() )
 			header('Content-type: text/javascript');
+		
+		$post_nonce = filter_input( INPUT_POST, 'nonce' );
 
-		if ( empty( $_POST['nonce'] ) || ! wp_verify_nonce($_POST['nonce'], 'carousel_nonce') )
+		if ( empty( $post_nonce ) || ! wp_verify_nonce($post_nonce, 'carousel_nonce') )
```

Comment:
Again, here `empty` was used to avoid notices when `$_POST['nonce']` is not set, so it should be enough to just pass the `filter_input` result into `wp_verify_nonce`:
``` 
if ( ! wp_verify_nonce( filter_input( INPUT_POST, 'nonce' ),  'carousel_nonce' ) )
```